pkg/cache/task.go

```diff
@@ -236,6 +238,36 @@ func (task *Task) isTerminated() bool {
 	return false
 }
 
+// task object initialization
+// normally when task is added, the task state is New
+// but during recovery, we need to init the task state according to
+// the task pod status. if the pod is already terminated,
+// we should mark the task as completed according.
+func (task *Task) initialize() {
+	task.lock.Lock()
+	defer task.lock.Unlock()
+
+	// task needs recovery means the task has already been
+	// scheduled by us with an allocation, instead of starting
+	// from New, directly set the task to Allocated.
+	if utils.NeedRecovery(task.pod) {
+		log.Logger().Info("init allocated task")
```

Comment:
Here in the log message we should include the at least the allocation UID,for better troubleshooting

Reply:
Added some more logs